dask/dataframe/categorical.py

```diff
@@ -27,15 +27,17 @@ def categorize(df, columns=None, **kwargs):
     if columns is None:
         dtypes = df.dtypes
         columns = [name for name, dt in zip(dtypes.index, dtypes.values)
-                    if dt == 'O']
+                   if dt == 'O']
     if not isinstance(columns, (list, tuple)):
         columns = [columns]
 
     distincts = [df[col].drop_duplicates() for col in columns]
     values = compute(*distincts, **kwargs)
 
     func = partial(_categorize_block, categories=dict(zip(columns, values)))
-    return df.map_partitions(func, columns=df.columns)
+
+    meta = func(df._pd)
+    return df.map_partitions(func, columns=meta)
```

Comment:
Request that we eventually change `columns=` to `meta=` or some other word like empty, sample, dummy, etc..